src/main/java/htsjdk/samtools/util/SnappyLoader.java

```diff
@@ -38,73 +39,71 @@
+                logger.debug("Snappy successfully loaded.");
             }
             /*
              * ExceptionInInitializerError: thrown by Snappy if native libs fail to load.
              * IllegalStateException: thrown within the `test.write` call above if no UTF-8 encoder is found.
              * IOException: potentially thrown by the `test.write` and `test.close` calls.
              * SnappyError: potentially thrown for a variety of reasons by Snappy.
              */
-            catch (ExceptionInInitializerError|IllegalStateException|IOException|SnappyError e) {
-                if (verbose) logger.warn("Snappy native library failed to load: " + e.getMessage());
+            catch (final ExceptionInInitializerError | IllegalStateException | IOException | SnappyError e) {
+                logger.warn(e, "Snappy native library failed to load.");
             }
-            SnappyAvailable = tmpSnappyAvailable;
+            snappyAvailable = tmpSnappyAvailable;
         }
     }
 
     /** Returns true if Snappy is available, false otherwise. */
-    public boolean isSnappyAvailable() { return SnappyAvailable; }
+    public boolean isSnappyAvailable() { return snappyAvailable; }
 
-    /** Wrap an InputStream in a SnappyInputStream. If Snappy is not available will throw an exception. */
+    /**
+     * Wrap an InputStream in a SnappyInputStream.
+     * @throws SAMException if Snappy is not available will throw an exception.
+     */
     public InputStream wrapInputStream(final InputStream inputStream) {
-        if (isSnappyAvailable()) {
-            try {
-                return new SnappyInputStream(inputStream);
-            } catch (Exception e) {
-                throw new SAMException("Error instantiating SnappyInputStream", e);
-            }
-        } else {
-            throw new SAMException("Snappy not available");
-        }
+        return wrapWithSnappyOrThrow(inputStream, SnappyInputStream::new);
     }
 
-    /** Wrap an InputStream in a SnappyInputStream. If Snappy is not available will throw an exception. */
+    /**
+     * Wrap an OutputStream in a SnappyOutputStream.
+     * @throws SAMException if Snappy is not available
+     */
     public OutputStream wrapOutputStream(final OutputStream outputStream) {
+        return wrapWithSnappyOrThrow(outputStream, (stream) -> new SnappyOutputStream(stream, SNAPPY_BLOCK_SIZE));
+    }
+
+    private interface IOFunction<T,R> {
+        R apply(T input) throws IOException;
+    }
+
+    private <T,R> R wrapWithSnappyOrThrow(T stream, IOFunction<T, R> wrapper){
         if (isSnappyAvailable()) {
             try {
-                return new SnappyOutputStream(outputStream, SNAPPY_BLOCK_SIZE);
+                return wrapper.apply(stream);
             } catch (Exception e) {
-                throw new SAMException("Error instantiating SnappyOutputStream", e);
+                throw new SAMException("Error instantiating SnappyInputStream", e);
```

Comment:
This should probably just read `Error wrapping stream with snappy` or something similarly generic.

Reply:
oh, good point, will change.